Library/Homebrew/extend/os/mac/resource.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Resource
+  class << self
+    undef on_macos
+
+    def on_macos(&_block)
+      yield
+    end
+  end
+end
```

Comment:
```suggestion
class Resource
  undef on_macos

  def on_macos(&_block)
    yield
  end
end
```

`class << self` is used for replacing class methods rather than instance methods.

I'm not awake enough to know how it worked for `Formula`.